crates/rune/src/ast/path.rs

```diff
@@ -49,6 +52,55 @@ impl Peek for Path {
     }
 }
 
+/// Parsing Paths
+///
+/// # Examples
+///
+/// ```rust
+/// use rune::{parse_all, ast, ParseError};
+///
+/// parse_all::<ast::Path>("x").unwrap();
+/// parse_all::<ast::Path>("::x").unwrap();
+/// parse_all::<ast::Path>("a::b").unwrap();
+/// parse_all::<ast::Path>("::ab::cd").unwrap();
+/// parse_all::<ast::Path>("crate").unwrap();
+/// parse_all::<ast::Path>("super").unwrap();
+/// parse_all::<ast::Path>("crate::foo").unwrap();
+/// parse_all::<ast::Path>("super::bar").unwrap();
+/// parse_all::<ast::Path>("::super").unwrap();
+/// parse_all::<ast::Path>("::crate").unwrap();
+/// ```
+///
+impl Parse for Path {
```

Comment:
IIUC, adding a `PathSegment` enum (as mentioned earlier) would allow for using a generated `Parse` impl again.